bin/check-immutability.py

```diff
@@ -0,0 +1,34 @@
+#!/usr/bin/env python 
+
+import json
+import subprocess
+import sys
+
+oldf = sys.argv[1]
+newf = sys.argv[2]
+
+immutable_keys = ('property', 'input', 'expected')
+
+# Use jq to flatten the test data, and parse it
+old = json.loads(subprocess.run([f"jq -r '[.. | objects | select(.uuid != null)]' {oldf}"], stdout=subprocess.PIPE, shell=True).stdout.decode('utf-8'))
+new = json.loads(subprocess.run([f"jq -r '[.. | objects | select(.uuid != null)]' {newf}"], stdout=subprocess.PIPE, shell=True).stdout.decode('utf-8'))
+
+# Convert new to dict uuid => case
+new = {c['uuid']: c for c in new}
+
+fails = set()
+
+# Iterate through old cases as only those could potentially be mutated
+for case in old:
+    for k in immutable_keys:
+        if case[k] != new[case['uuid']][k]:
+            fails.add(case['uuid'])
+            break
```

Comment:
Just to be sure: does this fail when the new file has removed one of the old test cases?

Reply:
It should because the dict key won't exist, but I'll test to be sure.

Reply:
Great!

Reply:
It already worked because it threw an error, but b010ec0 adds better handling. Sample output for a file that has been modified and deleted:

```
$ python3 bin/check-immutability.py test/canonical-data.json test/canonical-data-mutated-and-deleted.json

The following tests contain illegal mutations:
 - 9f98e133-eb7f-45b0-9676-cce001cd6f7a (Real part of a purely real number)
 - 07988e20-f287-4bb7-90cf-b32c4bffe0f3 (Real part of a purely imaginary number)
The following tests have been deleted illegally:
 - 657c55e1-b14b-4ba7-bd5c-19db22b7d659
```